resolver/resolver.cc

```diff
@@ -661,31 +661,63 @@ class ResolveConstantsWalk {
             return;
         }
 
-        if (send->args.empty()) {
-            return; // The arity mismatch error will be emitted later by infer.
-        }
+        auto *block = ast::cast_tree<ast::Block>(send->block);
 
-        for (auto &arg : send->args) {
-            auto argLoc = core::Loc(todo.file, arg.loc());
-            auto *id = ast::cast_tree<ast::ConstantLit>(arg);
+        if (!send->args.empty()) {
+            if (auto e = gs.beginError(loc, core::errors::Resolver::InvalidRequiredAncestor)) {
+                e.setHeader("`{}` only accepts a block", send->fun.show(gs));
+
+                string replacement = "";
+                int index = 0;
+                for (auto &arg : send->args) {
+                    string indent = "";
+                    if (index > 0) {
+                        auto locDetail = core::Loc::offset2Pos(todo.file.data(gs), send->loc.beginPos());
+                        for (int i = 1; i < locDetail.column; i++) {
+                            indent += " ";
+                        }
+                    }
 
-            if (id == nullptr || !id->symbol.exists() || !id->symbol.data(gs)->isClassOrModule()) {
-                if (auto e = gs.beginError(argLoc, core::errors::Resolver::InvalidRequiredAncestor)) {
-                    e.setHeader("Argument to `{}` must be statically resolvable to a class or a module",
-                                send->fun.show(gs));
+                    auto argLoc = core::Loc(todo.file, arg.loc());
+                    string argString = argLoc.source(gs).value();
+                    string ln = index < send->args.size() - 1 ? "\n" : "";
+
+                    replacement += fmt::format("{}{} {{ {} }}{}", indent, send->fun.show(gs), argString, ln);
```

Comment:
If the call had many arguments such as `requires_ancestor A, B`, we need to introduce a new line and adjust the indentation.